### PR TITLE
Adds skip snippet and scanoss setting feature

### PR DESCRIPTION
**Pull request by agustingroh**
### **What**
- Adds skip snippet flag
- Refactor on Policies 
- Adds scanoss settings (Export undeclared results as scanoss settings when scanossSettings flag is enabled)
- Upgrades scanoss-py version to v1.18.0
- Adds integration tests

### CODE REVIEW

File: codescantask/app.input.ts [hunk 1]
```diff
@@ -37,4 +37,9 @@ export const SBOM_ENABLED = tl.getInput('sbomEnabled') === 'true';
 export const SBOM_FILEPATH = tl.getInput('sbomFilepath') || "sbom.json" ;
 export const SBOM_TYPE = tl.getInput('sbomType');
 export const POLICIES_HALT_ON_FAILURE = tl.getInput('policiesHaltOnFailure') === 'true';
-export const RUNTIME_CONTAINER = tl.getInput('runtimeContainer') || "ghcr.io/scanoss/scanoss-py:v1.15.0";
+export const RUNTIME_CONTAINER = tl.getInput('runtimeContainer') || "ghcr.io/scanoss/scanoss-py:v1.18.0";
+export const SKIP_SNIPPETS = tl.getInput('skipSnippets') === 'true';
+export const SCAN_FILES = tl.getInput('scanFiles') === 'true';
+export const SCANOSS_SETTINGS = tl.getInput('scanossSettings') === 'true';
```
Comment (matiasdaloia) [inline codescantask/app.input.ts:43]: Here maybe we should do a check for the RUNTIME_CONTAINER

We should only allow to pass `SCANOSS_SETTINGS` and `SETTINGS_FILE_PATH` if the scanoss.py version supports it

File: codescantask/policies/undeclared-policy-check.ts [hunk 2]
```diff
@@ -42,66 +38,32 @@ export class UndeclaredPolicyCheck extends PolicyCheck {
...
-
-        if (nonDeclaredComponents.length === 0) {
-            await this.success(summary, details);
-        } else {
-            await this.reject(summary, details);
-        }
+    private buildArgs(): Array<string> {
+        return ['run', '-v', `${REPO_DIR}:/scanoss`, RUNTIME_CONTAINER, 'inspect', 'undeclared', '--input',
+            OUTPUT_FILEPATH, '--format', 'md',
+            ...(!SCANOSS_SETTINGS ? ['--sbom-format', 'legacy']: []) // Sets sbom format output to legacy if SCANOSS_SETTINGS is not true
```
Comment (matiasdaloia) [inline codescantask/policies/undeclared-policy-check.ts:45]: What happens if default `SBOM_FILEPATH` does not exists and `SCANOSS_SETTINGS` is false? We should pass `--sbom-format` anyways?

File: codescantask/app.input.ts (hunk 1 above)
Comment (agustingroh) [reply codescantask/app.input.ts:43]: Every Azure task has pingued an scanoss-py version. We shouldn´t have to check for the RUNTIME_CONTAINER

File: codescantask/policies/undeclared-policy-check.ts (hunk 2 above)
Comment (agustingroh) [reply codescantask/policies/undeclared-policy-check.ts:45]: If the SBOM_FILEPATH has a default value set. If the file is not found and the SCANOSS_SETTINGS is set to false, we use the legacy sbom format

**Review by eeisegn — APPROVED** [review]

**Pull request merged**